GVFS/GVFS.Common/MacProductUpgraderPlatformStrategy.cs

```diff
@@ -0,0 +1,62 @@
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System;
+using System.IO;
+
+namespace GVFS.Common
+{
+    public class MacProductUpgraderPlatformStrategy : ProductUpgraderPlatformStrategy

```

Comment:
Still looking but I'm not sure what `PlatformStrategy` means.  

I also don't think we should have platform specific code going into `GVFS.Common`.  Looking at this and it seems these should be a part of `IPlatformFileSystem` with a `TryPrepareDirectory(string path, out string error)`.  We might even be able to remove the 2 permission method from that interface since they appear only to be for Windows.
```
        bool TryCreateDirectoryWithAdminAndUserModifyPermissions(string directoryPath, out string error);
        bool TryCreateOrUpdateDirectoryToAdminModifyPermissions(ITracer tracer, string directoryPath, out string error);
```

Reply:
> it seems these should be a part of IPlatformFileSystem with a TryPrepareDirectory(string path, out string error)

I like this idea - but the problem I ran into is that that action depends on the directory (which the upgrader has the most context on).

Reply:
If there is an appropriate abstraction here, I am open to adding these directly to the `IPlatformFileSystem` class, but I haven't figured out a non-upgrader specific abstraction here... I will continue to think about this.